javagen/src/main/java/com/azure/autorest/Javagen.java

```diff
@@ -333,6 +335,23 @@ protected void writeClientModels(Client client, JavaPackage javaPackage, JavaSet
         }
     }
 
+    protected void writeHelperClasses(Client client, JavaPackage javaPackage, JavaSettings settings) {
+        // While azure-core's ResponseError hasn't shipped implementing JsonSerializable add a utility method that
+        // will serialize and deserialize ResponseError.
```

Comment:
Is there an issue tracking this to add support in core?

Reply:
I think Alan already merged the code in azure-core https://github.com/Azure/azure-sdk-for-java/pull/37046

@alzimmermsft After new azure-core get released, are we able to direct depends on azure-core to handle the de-serialization?

Reply:
Yeah, I've already merged in a change to `azure-core` where `ResponseError` now implements JsonSerializable so this won't be an issue any longer once that ships.